### DISCUSSION TITLE
HTMLResponse with attachment only displays html code instead of Download button

### DISCUSSION THREAD

**Discussion by magret-de-canard** [fastapi/fastapi#9628] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.responses import HTMLResponse

app = FastAPI()

@app.post("/test/", response_class=HTMLResponse)
    async def read_items():
        html_content = """
        <html>
            <head>
                <title>Some HTML in here</title>
            </head>
            <body>
                <h1>Look ma! HTML!</h1>
            </body>
        </html>
        """
        response = HTMLResponse(content=html_content, status_code=200)
        response.headers[
            "Content-Disposition"
        ] = f"attachment; filename=test.html"
        return response
```

### Description

Hi everyone,
I use FastAPI to generate a downloadable HTML file that is generated by a custom post function, the function returns an HTMLResponse, I set the header Content-Disposition to attachment and specified the name of the file that will be downloaded.

It worked perfectly until recently, when I executed the function from the /docs page, the Response body showed a clickable "Download file" button that allowed to download the generated file, which is the behavior I need.
But since recently, with no appearing reason, the function only displays the html code in the response body instead of the download button... 

I made here a minimal example of how to reproduce the problem, and here is what I see when I execute it on the docs :
![image](https://github.com/tiangolo/fastapi/assets/135727172/f30fa259-0edd-4522-95cf-f5a1e1718f3b)
We can see that content-disposition and media-type are set but no download button anymore, only the raw text html code...

I use fastapi 0.89.1 and python 3.10.0 and changed neither recently and I just can't figure out why the behavior of the API suddenly changed. I tried using different browser, didn't work, I tried using FileResponse instead by actually creating the file and giving its path to the FileResponse, didn't work either... 

I hope you can help me and I thank you a lot in advance if you can.

### Operating System

Linux

### FastAPI Version

0.89.1

### Python Version

Python 3.10.0

**Comment by chrisK824** [comment, marked as answer]
Hi @magret-de-canard , can you try adding a content type into your response headers like that:
``response.headers['Content-Type'] = "text/html"``
If the above doesn't work, you can try:
`response.headers['Content-Type'] = "application/octet-stream"`

**Reply by magret-de-canard** [reply]
Thanks a lot @chrisK824 ! Setting the content-type header to "application/octet-stream" worked. Have you any idea of a reason why it worked perfectly without it before, and now this setting is necessary, even though my version of fastAPI and its dependencies remained the same ?

**Reply by chrisK824** [reply]
You must have updated fastAPI at some point. I ve faced the same issue when I made some dependencies update and my export of some CSV files stopped behaving the same, just like what you witnessed. Didn't Search more into it as per what change caused this exactly since it came back with the content type header

**Reply by magret-de-canard** [reply]
It seems strange as my API was running in a docker container deployed on a k8s cluster, and the docker image installed everything from a requirements.txt file with specified versions that never changed according to my gitlab history. But whatever, as long as it works now. Thanks again !

**Reply by nymous** [reply]
Just chiming in, it could also be that Swagger UI deployed a new version. The default FastAPI docs load the swagger-ui JS directly from CDN without pinning a specific version, so it can change at any time.
(Not saying that's what happened, I couldn't find any reference for such change)

**Reply by jgould22** [reply]
It "pins" it on version 4, but only the major version.

https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/docs.py#L20
